### ISSUE TITLE
G-DINO to ONNX

### ISSUE THREAD

**Issue by kobic8** [IDEA-Research/GroundingDINO#156]
is there a way to export the Grounding DINO part to ONNX?

**Comment by oylz** [comment]
[here](https://github.com/IDEA-Research/GroundingDINO/issues/46#issuecomment-1611588814)

**Comment by kobic8** [comment]
> [here](https://github.com/IDEA-Research/GroundingDINO/issues/46#issuecomment-1611588814)

is there a code for that?

**Comment by wangshuang-jiayou** [comment]
> [here](https://github.com/IDEA-Research/GroundingDINO/issues/46#issuecomment-1611588814)

![image](https://github.com/IDEA-Research/GroundingDINO/assets/112836255/b63c200c-2a30-447f-a72c-295a2f1d65b3)
Hi~I have infer the onnx you provide,but got  negative logits. Is it normoal?

**Comment by oylz** [comment]
> > [here](https://github.com/IDEA-Research/GroundingDINO/issues/46#issuecomment-1611588814)
> 
> ![image](https://user-images.githubusercontent.com/112836255/250715206-b63c200c-2a30-447f-a72c-295a2f1d65b3.png) Hi~I have infer the onnx you provide,but got negative logits. Is it normoal?

yes.the same as the original model's result.

**Comment by wangshuang-jiayou** [comment]
> 

Thanks for your replay!!!I found the reason: I forgot to sigmoid the logits result. But got a little bit low class confidence~Could you provide the code? THANK YOU VERY MUCH~~~

**Comment by oylz** [comment]
> > > > [here](https://github.com/IDEA-Research/GroundingDINO/issues/46#issuecomment-1611588814)
> > > 
> > > 
> > > ![image](https://user-images.githubusercontent.com/112836255/250715206-b63c200c-2a30-447f-a72c-295a2f1d65b3.png) Hi~I have infer the onnx you provide,but got negative logits. Is it normoal?
> > 
> > 
> > yes.the same as the original model's result.
> 
> Thanks for your replay!!!I found the reason: I forgot to sigmoid the logits result. But got a little bit low class confidence~Could you provide the code? THANK YOU VERY MUCH~~~

the sigmoid call is goes like this
```
float sigmoid = 1.00/(1 + exp(-logits));
```

**Comment by wangshuang-jiayou** [comment]
> [here](https://github.com/IDEA-Research/GroundingDINO/issues/46#issuecomment-1611588814)

tensorrt doesn't support int64. Does the int64 in onnx casted to int32 in tensorrt influence the result in your experiment? My onnx results is right,but tensorrt result is wrong

**Comment by ganliqiang** [comment]
> > > > [here](https://github.com/IDEA-Research/GroundingDINO/issues/46#issuecomment-1611588814)
> > > 
> > > 
> > > ![image](https://user-images.githubusercontent.com/112836255/250715206-b63c200c-2a30-447f-a72c-295a2f1d65b3.png) Hi~I have infer the onnx you provide,but got negative logits. Is it normoal?
> > 
> > 
> > yes.the same as the original model's result.
> 
> Thanks for your replay!!!I found the reason: I forgot to sigmoid the logits result. But got a little bit low class confidence~Could you provide the code? THANK YOU VERY MUCH~~~

do you infer the onnx with tensorrt ?can you show your code? @wangshuang-jiayou

**Comment by QzYER** [comment]
> > [here](https://github.com/IDEA-Research/GroundingDINO/issues/46#issuecomment-1611588814)
> 
> tensorrt doesn't support int64. Does the int64 in onnx casted to int32 in tensorrt influence the result in your experiment? My onnx results is right,but tensorrt result is wrong

hello big shot，can you share your inference onnx scripts？thinks！

**Comment by QzYER** [comment]
> > [here](https://github.com/IDEA-Research/GroundingDINO/issues/46#issuecomment-1611588814)
> 
> tensorrt doesn't support int64. Does the int64 in onnx casted to int32 in tensorrt influence the result in your experiment? My onnx results is right,but tensorrt result is wrong

大佬你的tensorrt转换脚本能分享一下吗？

**Comment by Thivish16** [comment]
Hi,
It would be more helpful if you guys share the conversion logic that incorporated for converting to .onnx. I am now bit confused on the inference with onnx runtime. Thanks for the help!

**Comment by blacksino** [comment]
> > > > > [here](https://github.com/IDEA-Research/GroundingDINO/issues/46#issuecomment-1611588814)
> > > > 
> > > > 
> > > > ![image](https://user-images.githubusercontent.com/112836255/250715206-b63c200c-2a30-447f-a72c-295a2f1d65b3.png) Hi~I have infer the onnx you provide,but got negative logits. Is it normoal?
> > > 
> > > 
> > > yes.the same as the original model's result.
> > 
> > 
> > Thanks for your replay!!!I found the reason: I forgot to sigmoid the logits result. But got a little bit low class confidence~Could you provide the code? THANK YOU VERY MUCH~~~
> 
> do you infer the onnx with tensorrt ?can you show your code? @wangshuang-jiayou

你好，请问你解决这个问题了么，我转成onnx发现数值对不上

**Comment by Ranclus** [comment]
Good afternoon. Could some kind person please share the ONNX model or the code to perform the conversion?
